janus_core/single_point.py

```diff
@@ -72,44 +73,45 @@ def set_calculator(
             read_kwargs = read_kwargs if read_kwargs else {}
             self.read_system(**read_kwargs)
 
-        if isinstance(self.sys, list):
+        elif isinstance(self.sys, list):
```

Comment:
This may have resulted in an interesting error if `sys` was none. Maybe shouldn't be fixed here.

Reply:
shall we add a test for it?

Reply:
Ah, ok. Didn't realise that about `read_system` hence they may. mypy was complaining about potential NoneType